spyder/plugins/layout/plugin.py

```diff
@@ -931,7 +947,7 @@ def restore_visible_plugins(self):
 
         # Restore visible plugins
         for plugin in visible_plugins:
-            plugin_class = self.get_plugin(plugin)
+            plugin_class = self.main.get_plugin(plugin, error=False)
```

Comment:
Maybe the addition of the kwarg `error` to `get_plugin` method of the plugin class could be worthy instead of calling `self.main` directly here?